src/frame/api.js

```diff
@@ -0,0 +1,63 @@
+import runAdAuction from './auction/index.js';
+import joinAdInterestGroup from './interest-groups/join.js';
+import leaveAdInterestGroup from './interest-groups/leave.js';
+import renderAd from './render/index.js';
+
+export default async function fledgeAPI ({ data, ports }) {
+	try {
+		if (!Array.isArray(data)) {
+			throw new Error('data is not what it should be');
+		}
+
+		switch (data[0]) {
+			case 'joinAdInterestGroup': {
+				const [ , request ] = data;
+				const [ options, expiry ] = request;
```

Comment:
given there is an implied structure for `data`, what about making it more explicit?

instead of 
```
data = [ functionName, [ param1, param2, ...]];
```
how about
```
data = {
     method: functionName,
     params: [ param1, param2 ] // or possibly even structuring these further?
};
```

Reply:
I guess I could.  This aligns more with how the Google folks were doing things.

Reply:
Its good practice to understand why they are doing it this way when it seems at first glance a bit convoluted. It does appear that there are some array methods in JS that are more performant with postMessage than if using an object. It's unclear if those will matter with the size of data we are passing but this structure seems like best practice.